odl/discr/discr_mappings.py

```diff
@@ -687,42 +790,185 @@ def _evaluate(self, indices, norm_distances, out=None):
             return self.values[idx_res]
 
 
-class _NearestMeshgridInterpolator(_MeshgridInterpolator,
-                                   _NearestPointwiseInterpolator):
+def _compute_nearest_weights_edge(idcs, ndist, variant):
+    """Helper for nearest interpolation mimicing the linear case."""
+    # Get out-of-bounds indices from the norm_distances. Negative
+    # means "too low", larger than or equal to 1 means "too high"
+    lo = np.where(ndist < 0)
+    hi = np.where(ndist > 1)
+    ones = np.ones_like(ndist)
+    zeros = np.zeros_like(ndist)
+
+    # For "too low" nodes, the lower neighbor gets weight zero;
+    # "too high" gets 1.
+    if variant == 'left':
+        w_lo = np.where(ndist <= 0.5, ones, zeros)
+    else:
+        w_lo = np.where(ndist < 0.5, ones, zeros)
+
+    w_lo[lo] = 0
+    w_lo[hi] = 1
+
+    # For "too high" nodes, the upper neighbor gets weight zero;
+    # "too low" gets 1.
+    if variant == 'left':
```

Comment:
Possibly, and I can certainly see where this comes from mathematically, but I have a hard time seeing this working in practice since we always need to assume floats are approximate.